trl/trainer/bco_trainer.py

```diff
@@ -1290,7 +1291,7 @@ def _get_train_sampler(self) -> Optional[torch.utils.data.Sampler]:
             return None
         return SequentialSampler(self.train_dataset)
 
-    def get_batch_samples(self, model, batch: Dict[str, torch.LongTensor]) -> Tuple[str, str]:
+    def generate_from_model_and_ref(self, model, batch: Dict[str, torch.LongTensor]) -> Tuple[str, str]:
```

Comment:
Note for future: now that we have `LogCompletions` callback, it might be possible to enable the generative aspects of this method directly as a callback. We'd probably have to extend the `LogCompletions` callback to check if a reference model exists and generate for that too, but that seems better than having this code duplicated all over our preference trainers